config/v1/types_feature.go

```diff
@@ -131,6 +131,11 @@ type FeatureGateAttributes struct {
 	// +required
 	Name FeatureGateName `json:"name"`
 
+	// RequiredMinimumKubeletVersion is the lowest version the MinimumKubeletVersion field in the
+	// nodes.config object may be set to to enable this feature.
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
+	RequiredMinimumKubeletVersion string `json:"requiredMinimumKubeletVersion,omitempty"`
```

Comment:
Would prefer we make this future extensible, by making this a list type with a map key
```
requiredMinimumComponentVersion:
- component: Kubelet
  minimumVersion: 1.30.0
```
This would mean in the future we can add other minimum required components e.g. boot images, KAS etc if we find a need for it in the future

This would prevent people proxying the version through an association with kubelet and allow specific operators to act upon their minimum for their component

Reply:
Field needs to be marked as optional, with map key of component, component should be an enum type with a single valid value for now, maximum length of the list is 1

Reply:
updated!